### ISSUE TITLE
feature request:提供freeMarker模板nested语法类似的功能

### ISSUE THREAD

**Issue by zuomingcai** [vuejs/vue#4153]
freeMarker模板引擎支持的nested功能非常强大好用，希望vuejs提供类似的功能；
例如list组件render item时，用户可能对展示方式的要求各不相同，最好的方式是组件能让使用者自己指定item template，希望提供如下类似实现：
in child:

``` html
<template>
    <div v-for="item in list">
         <callback name="listItemCallback" context="item"/>
   </div>
</template>
```

in parent:

``` html
<template>
    <child :list="list">
         <callbackTemplate name="listItemCallback" context="item">
                 <img :src="item.url"/>< span>{{item.title}}< /span>  //此处不要有单根限制
         </callbackTemplate>
   </child>
<template>
```
如果不能支持，是否有其他的方法实现相同的需求

**Comment by posva** [comment]
There's something similar on the roadmap: **scoped slots** but I doubt it includes the for loop scoping.
FYI you can already achieve that with render functions by other functions that return vdom as props

**Comment by HerringtonDarkholme** [comment]
This has been tracked in Vue2.1. For now you can use inline-template hacks.

https://github.com/HerringtonDarkholme/vue-advanced-programming/blob/master/template-slot/helper.js

**Comment by zuomingcai** [comment]
@HerringtonDarkholme  the hack does not work any more after updated to latest version.

**Comment by HerringtonDarkholme** [comment]
Please file the issue in the related repo.

**Comment by zuomingcai** [comment]
@HerringtonDarkholme  sorry,it works,my bug.

**Comment by zuomingcai** [comment]
@HerringtonDarkholme 
how to use the v-outlet & v-template in a recursive treenode. 
only the root node get rendered from the template.

	<div class="ez-tree-node">
		<div class="ez-tree-node-self" @click="onItemClick">
			<span class="ez-tee-node-icon" :class="{'expanded':data.expanded,'hasChildren':data.hasChildren()}"></span>
			<v-outlet source="nodeTemplate" :$ctx="data">
				<span class="ez-tee-node-content">{{data.title}}</span>
			</v-outlet>
		</div>

		<div class="ez-tree-node-sub" v-if="data.hasChildren()" :class="{'expanded':data.expanded}">
			<span class="ez-tree-node-indent"></span>
			<span class="ez-tree-node-children">
				<template v-for="node in data.getChildren()">
					<ez-tree-node :data="node" />
				</template>
			</span>
		</div>
	</div>
-----------------------------------------------------------------------------------
        <ez-tree-node :data="books">
            <v-template slot="nodeTemplate" inline-template>
                <span>
                    {{$ctx.title}}
                </span>
            </v-template>
        </ez-tree-node>

**Comment by HerringtonDarkholme** [comment]
You can ask question in the original repo. https://github.com/HerringtonDarkholme/vue-advanced-programming/issues

Vue's issue list is for bug report and feature request. Thanks!

**Comment by yyx990803** [comment]
Closing (implemented as scoped slots in 2.1)
